LottieDropper/Dropbox/DropboxDetailViewModel.m

```diff
@@ -35,26 +35,79 @@ - (void) downloadFile:(void (^) (void)) done {
 
 	NSFileManager *fileManager = [NSFileManager defaultManager];
 	NSURL *outputDirectory = [fileManager URLsForDirectory:NSDocumentDirectory inDomains:NSUserDomainMask][0];
-	_fileOnDisk = [outputDirectory URLByAppendingPathComponent:self.file.pathLower];
-
-	[[[self.client.filesRoutes downloadUrl:self.file.pathLower overwrite:YES destination:self.fileOnDisk] setResponseBlock:^(DBFILESFileMetadata * _Nullable response, DBFILESDownloadError * _Nullable routeError, DBRequestError * _Nullable error, NSURL * _Nonnull destination) {
-		if (response) {
-			NSData *data = [[NSFileManager defaultManager] contentsAtPath:[destination path]];
-			NSError *error;
-			self.json = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingAllowFragments error:&error];
-
-			if (error != nil) {
-				NSLog(@"%@", error);
-			}
-
-		} else {
-			_fileOnDisk = nil;
-			NSLog(@"%@\n%@\n", routeError, error);
-		}
-		done();
-	}] setProgressBlock:^(int64_t bytesDownloaded, int64_t totalBytesDownloaded, int64_t totalBytesExpectedToDownload) {
-		NSLog(@"%lld\n%lld\n%lld\n", bytesDownloaded, totalBytesDownloaded, totalBytesExpectedToDownload);
-	}];
+    
+    // Check for dropbox or local files.
+    if (self.file.pathLower != nil) {
+        _fileOnDisk = [outputDirectory URLByAppendingPathComponent:self.file.pathLower];
+    } else {
+        _fileOnDisk = [outputDirectory URLByAppendingPathComponent:self.file.name];
+    }
+	
+    Connectivity *connectivity = [[Connectivity alloc]init];
+    if ([connectivity IsConnectionAvailable]) {
+        // Internet connection available
+        [self downloadFileFromService:done];
+    }
+    else {
+        // Internet connection unavailable
+        NSLog(@"No internet connection available");
+        [self checkIfFileExists:fileManager done:done];
+    }
+}
+
+/// Check if file exists at self.fileOnDisk
+- (void) checkIfFileExists: (NSFileManager *) fileManager done:(void (^) (void)) done {
```

Comment:
💀 Dit is niet asynchroon en veel te moeilijk nu met een `block` gewoon een bool terug geven is voldoende in dit geval.

`isFileAvailableForFilemanager:` die een `Bool retourneerd is goed.